Src/SemanticComparison/ProxyGenerator.cs

```diff
@@ -203,21 +210,15 @@ private static void BuildMethodEquals(TypeBuilder type, FieldInfo equalsHasBeenC
             var methodAttributes = MethodAttributes.Public | MethodAttributes.Virtual | MethodAttributes.HideBySig;
             MethodBuilder method = type.DefineMethod("Equals", methodAttributes);
 
-            MethodInfo objectEquals = typeof(object).GetMethod(
+            MethodInfo objectEquals = typeof(object).GetTypeInfo().GetMethod(
                 "Equals",
-                BindingFlags.Instance | BindingFlags.Public | BindingFlags.NonPublic,
-                null,
-                new[] { typeof(object) },
-                null
-                );
+                new[] {typeof(object)}
+            );
 
-            MethodInfo equalityComparerEquals = typeof(IEqualityComparer).GetMethod(

```

Comment:
Notice a change here - now only public method will be matched. That should be fine as method is public.